lib/msal-node/src/response/ManagedIdentityTokenResponse.ts

```diff
@@ -28,6 +28,28 @@ export type ManagedIdentityTokenResponse = {
     token_type?: AuthenticationScheme;
 
     // error
-    message?: string; // will be converted to error
-    correlationId?: string; // will be converted to correlation_id
+
+    /*
+     * (Web/Function) App Service
+     * 500 errors can return this from all MI sources as well
+     */
+    message?: string;
+    correlationId?: string;
+
+    // IMDS, Azure Arc, Service Fabric (unconfirmed)
+    error?: string | ErrorObject;
+    error_description?: string;
+    error_codes?: Array<string>;
+    correlation_id?: string;
+    timestamp?: string;
+    trace_id?: string;
+};
+
+/*
+ * This is the only error property that exists for Cloud Shell
+ * It can also be the only thing App Service will return
```

Comment:
In case of App service we don't get this format of error response. Cloud shell is the only one with nested json error response.

Reply:
Look at row #14 in [this document](https://microsoft.sharepoint.com/teams/ADAL/_layouts/15/Doc.aspx?sourcedoc={0339bdd6-b5e5-4d94-b4c9-ea47127f5023}&action=edit&wd=target%28ID4S%2FMSALs%2FTechnical%2FMSI.one%7Ca2a57687-9a17-4c9e-86d1-60aa32ab5005%2F%28Web%5C%29%20App%20Service%7Cc6be58ff-26ae-408f-9373-c8dd1c40ede3%2F%29&wdorigin=NavigationUrl) of MI errors.